docs/data/material/getting-started/templates/checkout/PaymentForm.js

```diff
@@ -1,65 +1,124 @@
+
+  const handleCvvChange = (event) => {
+    const value = event.target.value.replace(/\D/g, '');
+    if (value.length <= 3) {
+      setCvv(value);
+    }
+  };
+
+  const handleExpirationDateChange = (event) => {
+    const value = event.target.value.replace(/\D/g, '');
+    const formattedValue = value.replace(/(\d{2})(?=\d{2})/, '$1/');
+    if (value.length <= 4) {
+      setExpirationDate(formattedValue);
+    }
+  };
+
   return (
     <React.Fragment>
-      <Typography variant="h6" gutterBottom>
-        Payment method
-      </Typography>
-      <Grid container spacing={3}>
-        <Grid item xs={12} md={6}>
-          <TextField
-            required
-            id="cardName"
-            label="Name on card"
-            fullWidth
-            autoComplete="cc-name"
-            variant="standard"
-          />
-        </Grid>
-        <Grid item xs={12} md={6}>
-          <TextField
-            required
-            id="cardNumber"
-            label="Card number"
-            fullWidth
-            autoComplete="cc-number"
-            variant="standard"
-          />
-        </Grid>
-        <Grid item xs={12} md={6}>
-          <TextField
-            required
-            id="expDate"
-            label="Expiry date"
-            fullWidth
-            autoComplete="cc-exp"
-            variant="standard"
-          />
-        </Grid>
-        <Grid item xs={12} md={6}>
-          <TextField
-            required
-            id="cvv"
-            label="CVV"
-            helperText="Last three digits on signature strip"
-            fullWidth
-            autoComplete="cc-csc"
-            variant="standard"
-          />
-        </Grid>
-        <Grid item xs={12}>
-          <FormControlLabel
-            control={<Checkbox color="secondary" name="saveCard" value="yes" />}
-            label="Remember credit card details for next time"
+      <Box
+        sx={{
+          display: 'flex',
+          justifyContent: 'center',
+          mb: 4,
+        }}
+      >
+        <Box
+          sx={{
+            display: 'flex',
+            flexDirection: 'column',
+            justifyContent: 'space-between',
+            p: 3,
+            height: { xs: '250px', sm: '350px' },
+            width: { xs: '375px', sm: '525px' },
+            borderRadius: '20px',
+            backdropFilter: 'blur(35px)',
+            border: '2px solid ',
+            borderColor: 'divider',
+            boxShadow: '0px 4px 8px rgba(0, 0, 0, 0.05)',
+          }}
+        >
+          <Box sx={{ display: 'flex', justifyContent: 'space-between' }}>
+            <Typography variant="subtitle2">Credit card</Typography>
+            <CreditCardRoundedIcon />
+          </Box>
+          <SimCardRoundedIcon
+            sx={{ fontSize: { xs: '48px', sm: '56px' }, transform: 'rotate(90deg)' }}
           />
-        </Grid>
-      </Grid>
+          <Box sx={{ display: 'flex', justifyContent: 'space-between', gap: 3 }}>
+            <TextField
```

Comment:
I understand why using a Text Field here, but it seems a bit inconsistent to use different types of inputs throughout the same template. I get why you used it here, as it mimics a credit card a bit more, but maybe it's worth exploring using the inputs from the previous step here, too.

Reply:
I'm not so convinced about it, what do you think?

<img width="508" alt="Screenshot 2024-02-28 at 11 45 46" src="https://github.com/mui/material-ui/assets/37222944/3e52ec5a-f0d2-484e-9a6d-d47ee21852a3">

Reply:
I think it'll look better if you add labels to the inputs (and also more accessible as well!). The previous Text Field is very Material Design-y, in my opinion 😬